src/resource/resource.js

```diff
@@ -3,8 +3,18 @@ import ResourceConfig from './config';
 /* A resource defines a contract for the inputs and outputs of Operations.
  */
 export default class Resource {
-    constructor(config) {
+    constructor(options = {}) {
+        const { config, registry } = options;
+
         this.config = config || new ResourceConfig();
+        this.registry = registry || {};
```

Comment:
I'm adding a registry to each resource. It's used locally in some cases, but not yet globally.